chef/cookbooks/salt/recipes/salt_ssh.rb

```diff
@@ -0,0 +1,55 @@
+# requires repos
+# http://download.nue.suse.com/ibs/SUSE/Products/SLE-Module-Adv-Systems-Management/12/x86_64/product/
+# http://download.nue.suse.com/ibs/SUSE/Updates/SLE-Module-Adv-Systems-Management/12/x86_64/update/
+package "salt-ssh" do
+  action :install
+end
+
+nodes = node_search_with_cache("roles:network")
+roster = []
+nodes.each do |n|
+  next if n[:dns].nil?
```

Comment:
better search for ```roles:dns-client OR roles:dns-server``` then (these are attributes coming from dns barclamp). just in case there is an inconsistency between network and dns barclamp, that avoids this extra next then.